docs/swift/DesignDoc.md

```diff
@@ -224,6 +245,16 @@ drafted or sent proposals for review that would benefit our designs:
 *   [Drafted: Resettable Properties]
     (https://github.com/patters/swift-evolution/blob/master/proposals/0000-resettable-properties.md)
 
+Lastly, we note some name transformations that we would apply to fields:
+
+* Repeated fields will have `Array` appended to their names. This is because
+  there is a trend in some messages to name repeated fields with the singular
+  version of the name (`descriptor.proto` in particular does this), and
+  `foo.name.append(bar)` looks potentially like appending a string to another
+  string whereas `foo.nameArray.append(bar)` is much clearer.
+* If a field name would collide with one of the properties or methods in
+  `ProtoMessage`'s public interface, we append `Field` to its name.
```

Comment:
Yea, when I updated things, I stuck with `Array` and not `_Array`.  Was just meaning to call out the issues.

It does detect something ending in _Array_ and adds the suffix so it hopefully won't appear like an array (since it isn't). [Code here.](https://github.com/google/protobuf/blob/master/src/google/protobuf/compiler/objectivec/objectivec_helpers.cc#L455)

Historically, my one concern with trying to wait until conflicts happen is that can mean a proto "works", people code to it, then they update the proto, and a generated property name changes, breaking their code.  When this happens, by definition, there is now a second method with a "similar" name, adding to the confusion on unbreaking the code.  So as much as possible, I'd tried to error on the behaviors not depending on the other fields in the message.